### PR TITLE
[Potentially breaking change] changed specs2 to single bind

### PR DESCRIPTION
**Pull request by ittaiz**
changed specs2 to single bind to remove volatility of version upgrade (dependency graph changes)
@johnynek following the discussion in #311 I realized my implementation was way off with the multiple binds.
This is a breaking change but I actually stumbled upon this need a few times in the past but I didn't realize the problem until your explanation in #311

### CODE REVIEW

**Comment by ittaiz** [comment]
@chaoren I needed to change the aspect test for this to pass.
Will the plugin work in the above configuration?
Specs2 dependencies are very volatile

**Comment by ittaiz** [comment]
Also this will be a breaking change only to users who've customized specs2 usage

**Comment by chaoren** [comment]
Yes, it looks fine. It's even better now that the aspect test doesn't need to be changed every single time the dependencies change.

**Comment by ittaiz** [comment]
Great.
If someone doesn't load the specs2_repositories and binds to its own target
which has different specs2 jars will the IDE be ok with it?

**Comment by chaoren** [comment]
It doesn't matter, since  `scala_specs2_junit_test` is just a macro on top of `scala_junit_test`, all the aspect sees is a `scala_junit_test` with a bunch of extra deps attached.

**Comment by ittaiz** [comment]
@johnynek I want to merge this one and #320 tomorrow since I have a backlog of stuff to upstream and just fix here:
1. jars2labels accumulation has bugs (push upstream)
2. Bazel 0.7.0 rules_Scala integration broken (don't know how to fix yet, not just a line change)
3. IntelliJ break (intellij head and rules_scala head broken as @gregestren shows in #309 )
4. scala_import rule (needed by many and is crucial for strict deps)
Due to our problematic monolith many changes need to be serial or accumulated to a big PR. Between the two I'd rather serialize them and try to have quick review cycles on small PRs.

**Comment by ittaiz** [comment]
@or-shachar wdyt?

File: specs2/BUILD [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      "@io_bazel_rules_scala_org_scalaz_scalaz_effect//jar",
+      "@io_bazel_rules_scala_org_scalaz_scalaz_core//jar",
+    ],
+    deps = [
+      "//external:io_bazel_rules_scala/dependency/scala/scala_xml",
+      "//external:io_bazel_rules_scala/dependency/scala/parser_combinators",
+      "//external:io_bazel_rules_scala/dependency/scala/scala_library",
+      "//external:io_bazel_rules_scala/dependency/scala/scala_reflect"
+    ]
+)
```
Comment (or-shachar) [inline specs2/BUILD]: [cosmetic] missing newline

File: specs2/specs2.bzl [hunk 2]
```diff
@@ -11,44 +11,32 @@ def specs2_repositories():
...
-  "//external:io_bazel_rules_scala/dependency/specs2/specs2_core",
-  "//external:io_bazel_rules_scala/dependency/specs2/specs2_common",
-  "//external:io_bazel_rules_scala/dependency/specs2/specs2_matcher",
-  "//external:io_bazel_rules_scala/dependency/scalaz/scalaz_effect",
-  "//external:io_bazel_rules_scala/dependency/scalaz/scalaz_core",
-  "//external:io_bazel_rules_scala/dependency/scala/scala_xml",
-  "//external:io_bazel_rules_scala/dependency/scala/parser_combinators",
-  "//external:io_bazel_rules_scala/dependency/scala/scala_library",
-  "//external:io_bazel_rules_scala/dependency/scala/scala_reflect"]  
+  return ["//external:io_bazel_rules_scala/dependency/specs2/specs2"]
```
Comment (or-shachar) [inline specs2/specs2.bzl]: [cosmetic] - misssing newline

File: specs2/BUILD (hunk 1 above)
Comment (ittaiz) [reply specs2/BUILD]: done

File: specs2/specs2.bzl (hunk 2 above)
Comment (ittaiz) [reply specs2/specs2.bzl]: done

**Pull request merged**